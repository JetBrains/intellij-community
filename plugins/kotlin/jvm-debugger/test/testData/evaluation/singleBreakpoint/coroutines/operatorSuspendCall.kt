
// ATTACH_LIBRARY: maven(org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.2)

class Foo {
    private val content = listOf(1, 2, 3)
    suspend operator fun iterator(): Iterator<Int> = content.iterator()
}

fun main() {
    val x = Foo()

    //Breakpoint!
    println()
}

// EXPRESSION: buildList { for (i in x) { add(i) } }.sum()
// RESULT: 6: I
