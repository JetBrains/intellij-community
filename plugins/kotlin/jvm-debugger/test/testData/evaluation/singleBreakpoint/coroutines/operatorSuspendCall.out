LineBreakpoint created at operatorSuspendCall.kt:13
Run Java
Connected to the target VM
operatorSuspendCall.kt:13
Compile bytecode for kotlinx.coroutines.runBlocking {
    buildList {
        for (i in x) {
            add(i)
        }
    }.sum()
}
Disconnected from the target VM

Process finished with exit code 0
